### PR TITLE
Migrate Resource List Pages from Static to Dynamic extensions

### PR DESCRIPTION
**Pull request by bipuladh**

### CODE REVIEW

File: frontend/public/components/resource-list.tsx [hunk 1]
```diff
@@ -57,11 +64,9 @@ export const ResourceListPage = connectToPlural(
           <title>{kindObj.labelPlural}</title>
         </Helmet>
         <AsyncComponent
-          autoFocus={!noProjectsAvailable}
           kind={modelRef}
           loader={componentLoader}
           match={props.match}
-          mock={noProjectsAvailable}
```
Comment (christianvogt) [inline frontend/public/components/resource-list.tsx:64]: Is this removal safe?

File: frontend/packages/console-dynamic-plugin-sdk/src/extensions/pages.ts [hunk 2]
```diff
@@ -15,7 +15,32 @@ export type StandaloneRoutePage = ExtensionDeclaration<
   }
 >;
 
+type ResourcePageCommonProps = {
+  /** See https://reacttraining.com/react-router/web/api/match */
+  match: RouteComponentProps['match'];
+  /** The resource kind scope. */
+  kind: string;
```
Comment (christianvogt) [inline frontend/packages/console-dynamic-plugin-sdk/src/extensions/pages.ts:22]: Shouldn't this be expanded to:
```
 model: {
  group: string,
  version: string,
  kind: string,
 }, 
```

Comment (bipuladh) [reply frontend/packages/console-dynamic-plugin-sdk/src/extensions/pages.ts:22]: I thought we would be passing GVK identifier to the Component.

Comment (christianvogt) [reply frontend/packages/console-dynamic-plugin-sdk/src/extensions/pages.ts:22]: The format of our single GVK string is an internal detail and we can construct it internally given the expanded object.

**Pull request closed without merging**